src/aws_encryption_sdk/keyrings/aws_kms/__init__.py

```diff
@@ -0,0 +1,376 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Keyring for use with AWS Key Management Service (KMS).
+
+.. versionadded:: 1.5.0
+
+"""
+import logging
+
+import attr
+import six
+from attr.validators import deep_iterable, instance_of, optional
+
+from aws_encryption_sdk.exceptions import DecryptKeyError, EncryptKeyError
+from aws_encryption_sdk.identifiers import AlgorithmSuite
+from aws_encryption_sdk.internal.validators import value_is_not_a_string
+from aws_encryption_sdk.keyrings.base import Keyring
+from aws_encryption_sdk.keyrings.multi import MultiKeyring
+from aws_encryption_sdk.materials_managers import DecryptionMaterials, EncryptionMaterials
+from aws_encryption_sdk.structures import EncryptedDataKey, KeyringTrace, KeyringTraceFlag, MasterKeyInfo, RawDataKey
+
+from .client_suppliers import ClientSupplier, DefaultClientSupplier
+
+try:  # Python 3.5.0 and 3.5.1 have incompatible typing modules
+    from typing import Any, Dict, Iterable, Union  # noqa pylint: disable=unused-import
+except ImportError:  # pragma: no cover
+    # We only actually need these imports when running the mypy checks
+    pass
+
+__all__ = ("KmsKeyring",)
+
+_LOGGER = logging.getLogger(__name__)
+_PROVIDER_ID = "aws-kms"
+_GENERATE_FLAGS = {KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY}
```

Comment:
Should this also have Encrypted and Signed?

Reply:
This is ok, for now.
This does not block this release,
and mirrors the Java implementation.